File: JVA/ACM/LinkedList.java
package JVA.ACM;

public class LinkedList {
    private ListNode head;
    LinkedList(int[] arr, boolean dummyHead) {
        if (arr == null || arr.length == 0) {
            throw new IllegalArgumentException("Array cannot be empty");
        }
        if (dummyHead) {
            head = new ListNode(-1);
            ListNode cur = head;
            for (int i = 0; i < arr.length; i++) {
                cur.next = new ListNode(arr[i]);
                cur = cur.next;
            }
        } else {
            head = new ListNode(arr[0]);
            ListNode cur = head;
            for (int i = 1; i < arr.length; i++) {
                cur.next = new ListNode(arr[i]);
                cur = cur.next;
            }
        }
    }

    public ListNode getHead() {
        return head;
    }

    //toString
    public String toString() {
        StringBuilder res = new StringBuilder();
        ListNode cur = head;
        while (cur != null) {
            res.append(cur.val + " -> ");
            cur = cur.next;
        }
        res.append("NULL");
        return res.toString();
    }

}

File: JS/ACM/linkedList.js
class Node {
  constructor(val) {
    this.val = val;
    this.next = null;
  }
}

export default class LinkedList {
  constructor(arr) {
    this.head = new Node(arr[0]);
    let cur = this.head;
    for (let i = 1; i < arr.length; i++) {
      cur.next = new Node(arr[i]);
      cur = cur.next;
    }
  }
  toString() {
    let cur = this.head;
    let str = "";
    while (cur) {
      str += cur.val + " -> ";
      cur = cur.next;
    }
    return str + "null";
  }
  getHead() {
    return this.head;
  }
}
